### PR TITLE
Let the SDL be passed into the service list rather than always fetching it from the URL

### PR DESCRIPTION
**Pull request by ThomasJamesCrawford**
I think this is a simple way to allow using a schema registry. The registry logic can be handled outside to supply the SDL, Url and Name of each service. Then just pass these in for each.

### CODE REVIEW

**Review by mcollina — COMMENTED** [review]
Good work! I think this should be expanded a bit as in https://github.com/mercurius-js/mercurius/blob/master/docs/federation.md#programmatically-refresh-federated-schemas-in-gateway-mode we can use `refresh()` to have all the schema updated. _Maybe_ we could also support it do be a function that returns the schema? Or maybe expose a `setSchema()` somewhere?

File: lib/gateway/service-map.js [hunk 1]
```diff
@@ -63,6 +63,38 @@ async function getWsOpts (service) {
...
+    }),
+    headers
+  })
+
+  const {
+    json: { data = null, error, message, statusCode }
+  } = response
+
+  if (data === null) {
+    throw Object.assign(new Error(), { error, message, statusCode })
```
[the remark this answers is not in the document]
Comment (ThomasJamesCrawford) [reply lib/gateway/service-map.js]: Not really sure if I'm using it correctly but I've given it a go

File: docs/federation.md [hunk 2]
```diff
@@ -252,6 +253,73 @@ setTimeout(async () => {
...
+            id: ID!
+            name: String
+          }
+        `
+      }
+    ]
+  }
+})
+
+server.listen(3002)
```
Comment (mcollina) [inline docs/federation.md:302]:
```suggestion
await server.listen(3002)
```

**Review by mcollina — COMMENTED** [review]
Good work! Just a nit

**Review by mcollina — APPROVED** [review]
lgtm

**Review by PabloSzx — APPROVED** [review]
looks good 👌 

PS: in Windows there are issues with the test snapshots of "test/routes.js", related to the end of line of the stringified functions of the playground headers ([screenshot](https://i.imgur.com/2AjZZdk.png)), but that is another minor issue.

**Comment by mcollina** [comment]
We might want to add Windows to our CI @PabloSzx, would you like to send a PR?

**Pull request merged**